src/main/java/org/mule/extension/http/api/listener/builder/HttpListenerSuccessResponseBuilder.java

```diff
@@ -89,13 +91,21 @@ public void setBody(TypedValue<Object> body) {
   }
 
   @Override
-  public MultiMap<String, String> getHeaders() {
-    return headers;
+  public CaseInsensitiveMultiMap getHeaders() {
+    return headers.toImmutableMultiMap();
   }
 
   @Override
   public void setHeaders(MultiMap<String, String> headers) {
-    this.headers = headers != null ? headers : emptyMultiMap();
+    if (headers != null) {
```

Comment:
i've seen this code before. Maybe there should be an utils (a static method in `CaseInsensitiveMultiMap`) that does this conversion